tests/sources/fixtures/confluence/fixture.py

```diff
@@ -20,16 +21,16 @@
 
 match DATA_SIZE:
     case "small":
-        SPACE_COUNT = 100
-        SPACE_OBJECT_COUNT = 100
+        SPACE_COUNT = 10
+        SPACE_OBJECT_COUNT = 25
         ATTACHMENT_COUNT = 3
     case "medium":
-        SPACE_COUNT = 100
-        SPACE_OBJECT_COUNT = 200
+        SPACE_COUNT = 10
+        SPACE_OBJECT_COUNT = 50
         ATTACHMENT_COUNT = 5
     case "large":
-        SPACE_COUNT = 100
-        SPACE_OBJECT_COUNT = 250
+        SPACE_COUNT = 10
+        SPACE_OBJECT_COUNT = 75
```

Comment:
This looks to me like a significant reduction in performance. Can you perform a benchmark comparison for this PR against the current Confluence connector?
If the performance hit is significant, we might want to consider making this optional through an RCF.

Reply:
Now we are indexing the documents space by space

If we calculate the number of docs to be indexed in functional test, the stats are as follows for the `medium` DATA_SIZE:
fixture with the older approach where spaces and other objects are getting indexed separately:
- `SPACE_COUNT + (SPACE_OBJECT_COUNT * ATTACHMENT_COUNT * 2)`
- Total docs - `100 + 200 * 5 * 2 = 2100`

whereas in updated fixture:
- `SPACE_COUNT + (SPACE_COUNT * SPACE_OBJECT_COUNT * ATTACHMENT_COUNT * 2)`
- Total docs - `10 + 10 * 50 * 5 * 2 = 5010`

So, eventually ftest is indexing more docs now.

Reply:
If we don't reduce the numbers in the fixture, it'll try to index `100 (spaces) + 100 (spaces) * 200 (page/blogpost) * 2 * 5 (attachment) = 2,00,010 doc` and so the CI will fail.

Reply:
@navarone-feekery we're preparing a performance doc, will attach it in the description.

Reply:
Thanks for the extra effort here, I'm no longer concerned about the performance impact.